upstairs/src/downstairs.rs

```diff
@@ -3261,6 +3261,18 @@ impl Downstairs {
         }
     }
 
+    /// Returns the number of clients that can accept IO
+    ///
+    /// A client can accept IO if it is in the `Active` or `LiveRepair` state.
+    pub fn active_client_count(&self) -> usize {
+        self.clients
+            .iter()
+            .filter(|c| {
+                matches!(c.state(), DsState::Active | DsState::LiveRepair)
```

Comment:
I think we need to allow additional states here.  `Offliine` for instance is one where we have not yet decided what the situation is, and it could just be a downstairs rebooting and soon enough (fingers crossed) it will come back.

There may be other transitory states to consider as well.

Reply:
Good point, I'll revisit this.  In parallel, I've been working on simplifying the state machine; see https://github.com/oxidecomputer/crucible/pull/1568 and https://github.com/oxidecomputer/crucible/pull/1570

Reply:
If this goes in before 1568/1570, then we want the updated list of `DsState`'s